src/app.ts

```diff
@@ -267,13 +303,15 @@ async function clearAndReloadSpidStrategy(
   let newSpidStrategy: SpidStrategy | undefined;
   try {
     // Inject a new SPID Strategy generate function inside the container
-    container.register(SPID_STRATEGY, defaultModule.loadSpidStrategy());
-    newSpidStrategy = await container.resolve(SPID_STRATEGY);
+    // tslint:disable-next-line: no-object-mutation
+    defaultModule.currentSpidStrategy = defaultModule.loadSpidStrategy();
```

Comment:
can we have `currentSpidStrategy` to be the result of the promise? is it feasible? i.e.:

```suggestion
    defaultModule.currentSpidStrategy = await defaultModule.loadSpidStrategy();
```